src/main/java/decodes/cwms/CwmsSqlDatabaseIO.java

```diff
@@ -304,4 +318,98 @@ public IntervalDAI makeIntervalDAO()
 		return new CwmsIntervalDAO(this, dbOfficeId);
 	}
 
+	@Override
+	public Connection getConnection()
+	{
+		// Called from DAOs and DbIo to get a new connection from the pool.
+		if (conInfo == null || conInfo.getLoginInfo() == null)
+		{
+			Logger.instance().failure(module + ".getConnection -- loginInfo is null! DB not initialized?");
+			return null;
+		}
+		
+		Connection ret = null;
+		try
+		{
+			ret = CwmsDbConnectionPool.getInstance().getConnection(conInfo.getLoginInfo());
+		}
+		catch (SQLException ex)
+		{
+			String msg = module + ".getConnection() Cannot get CWMS connection for user '"
+				+ conInfo.getLoginInfo().getUser() + "' and dbURI '" 
+				+ conInfo.getLoginInfo().getUrl() + "'";
+			Logger.instance().failure(msg);
+			PrintStream ps = Logger.instance().getLogOutput();
+			if (ps != null)
+				ex.printStackTrace(ps);
+			else
+				ex.printStackTrace(System.err);
+			return null;
+			// TODO Should this method throw BadConnectException?
```

Comment:
yes, if the whole pool is failing there's likely an unrecoverable database failure.